### ISSUE TITLE
`react/jsx-key` Does not report errors when child is optional

### ISSUE THREAD

**Issue by tylercraft** [jsx-eslint/eslint-plugin-react#3481]
This does not warn on this use case:

```
data.map(x => somecheck && <Hello>{x}</Hello>);
```
